src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/Contracts/SchemaCompareOpenScmpRequest.cs

```diff
@@ -0,0 +1,87 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using Microsoft.SqlTools.Hosting.Protocol.Contracts;
+using Microsoft.SqlTools.ServiceLayer.TaskServices;
+using Microsoft.SqlTools.ServiceLayer.Utility;
+using System.Collections.Generic;
+
+namespace Microsoft.SqlTools.ServiceLayer.SchemaCompare.Contracts
+{
+    public class SchemaCompareObjectId
+    {
+        /// <summary>
+        /// Name to create object identifier
+        /// </summary>
+        public string Name;
```

Comment:
string[] NameParts
to preserve the "." and "[" in name and avoid any string operations in sqltools service as we talked
(don't have to be in this PR)

Reply:
changed to NameParts